src/web/crate_details.rs

```diff
@@ -741,4 +754,42 @@ mod tests {
             Ok(())
         });
     }
+
+    #[test]
+    fn feature_flags_is_hidden_when_empty() {
+        wrapper(|env| {
+            env.fake_release()
+                .name("binary")
+                .version("0.1.0")
+                .binary(true)
+                .features(HashMap::new())
+                .create()?;
+
+            let page = kuchiki::parse_html()
+                .one(env.frontend().get("/crate/binary/0.1.0").send()?.text()?);
+            assert!(page.select_first(r#"a[aria-label="Feature"]"#).is_err());
+            Ok(())
+        });
+    }
+
+    #[test]
+    fn feature_private_feature_flags_are_hidden() {
+        wrapper(|env| {
+            let features = [("_private".into(), Vec::new())]
+                .iter()
+                .cloned()
+                .collect::<HashMap<String, Vec<String>>>();
```

Comment:
You use this later, so I think the type annotation can be removed.
```suggestion
                .collect()
```